Mathlib/RepresentationTheory/Action/Continuous.lean

```diff
@@ -0,0 +1,116 @@
+/-
+Copyright (c) 2024 Christian Merten. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christian Merten
+-/
+import Mathlib.RepresentationTheory.Action.Basic
+import Mathlib.Topology.Algebra.MulAction
+import Mathlib.Topology.Category.TopCat.Basic
+
+/-!
+
+# Topological subcategories of `Action V G`
+
+For a concrete category `V`, where the forgetful functor factors via `TopCat`,
+and a monoid `G`, equipped with a topological space instance,
+we define the full subcategory `ContAction V G` of all objects of `Action V G`
+where the induced action is continuous.
+
+We also define a category `DiscreteContAction V G` as the full subcategory of `ContAction V G`,
+where the underlying topological space is discrete.
+
+Finally we define inclusion functors into `Action V G` and `TopCat` in terms
+of `HasForget₂` instances.
+
+-/
+
+universe u v
+
+open CategoryTheory Limits
+
+variable (V : Type (u + 1)) [LargeCategory V] [ConcreteCategory V] [HasForget₂ V TopCat]
+variable (G : MonCat.{u}) [TopologicalSpace G]
+
+namespace Action
+
+instance : HasForget₂ (Action V G) TopCat :=
+  HasForget₂.trans (Action V G) V TopCat
+
+instance (X : Action V G) : MulAction G ((CategoryTheory.forget₂ _ TopCat).obj X) where
+  smul g x := ((CategoryTheory.forget₂ _ TopCat).map (X.ρ g)) x
+  one_smul x := by
+    show ((CategoryTheory.forget₂ _ TopCat).map (X.ρ 1)) x = x
+    simp
+  mul_smul g h x := by
+    show (CategoryTheory.forget₂ _ TopCat).map (X.ρ (g * h)) x =
+      ((CategoryTheory.forget₂ _ TopCat).map (X.ρ h)
+        ≫ (CategoryTheory.forget₂ _ TopCat).map (X.ρ g)) x
+    rw [← Functor.map_comp, map_mul]
+    rfl
+
+/-- For `HasForget₂ V TopCat` a predicate on an `X : Action V G` saying that the induced action on
+the underlying topological space is continuous. -/
+def IsContinuous (X : Action V G) : Prop :=
+  ContinuousSMul G ((CategoryTheory.forget₂ _ TopCat).obj X)
```

Comment:
As `ContinuousSMul` is a type-class, it could be nice to replace `def` by `abbrev` here (and set `variable {V G}`), which will some modifications below.